go/arrow/csv/common.go

```diff
@@ -239,21 +239,31 @@ func WithStringsReplacer(replacer *strings.Replacer) Option {
 
 func validate(schema *arrow.Schema) {
 	for i, f := range schema.Fields() {
-		switch ft := f.Type.(type) {
-		case *arrow.BooleanType:
-		case *arrow.Int8Type, *arrow.Int16Type, *arrow.Int32Type, *arrow.Int64Type:
-		case *arrow.Uint8Type, *arrow.Uint16Type, *arrow.Uint32Type, *arrow.Uint64Type:
-		case *arrow.Float16Type, *arrow.Float32Type, *arrow.Float64Type:
-		case *arrow.StringType, *arrow.LargeStringType:
-		case *arrow.TimestampType:
-		case *arrow.Date32Type, *arrow.Date64Type:
-		case *arrow.Decimal128Type, *arrow.Decimal256Type:
-		case *arrow.ListType, *arrow.LargeListType, *arrow.FixedSizeListType:
-		case *arrow.BinaryType, *arrow.LargeBinaryType, *arrow.FixedSizeBinaryType:
-		case arrow.ExtensionType:
-		case *arrow.NullType:
-		default:
-			panic(fmt.Errorf("arrow/csv: field %d (%s) has invalid data type %T", i, f.Name, ft))
+		if !typeSupported(f.Type) {
+			panic(fmt.Errorf("arrow/csv: field %d (%s) has invalid data type %T", i, f.Name, f.Type))
 		}
 	}
 }
+
+func typeSupported(dt arrow.DataType) bool {
+	switch dt := dt.(type) {
+	case *arrow.BooleanType:
+	case *arrow.Int8Type, *arrow.Int16Type, *arrow.Int32Type, *arrow.Int64Type:
+	case *arrow.Uint8Type, *arrow.Uint16Type, *arrow.Uint32Type, *arrow.Uint64Type:
+	case *arrow.Float16Type, *arrow.Float32Type, *arrow.Float64Type:
+	case *arrow.StringType, *arrow.LargeStringType:
+	case *arrow.TimestampType:
+	case *arrow.Date32Type, *arrow.Date64Type:
+	case *arrow.Decimal128Type, *arrow.Decimal256Type:
```

Comment:
should we bother actively listing these out? it might make more sense to just explicitly have cases for the types we don't support to return false and then just default return true?

just to shrink this

Reply:
I'd leave this out of scope for now

Reply:
But making an inclusive list might be a good idea, 100%

Reply:
that's fine. as long as the tests still pass and all is good, I'm fine with cleaning this up as a follow-up

Reply:
I'd say `no`, as this allows explicitly listing what's supported, & won't fail on the newly added types (string view, list view?)

Reply:
that makes sense, though it does mean we should probably add those to this and have them return false like we do for map and other types lol. but that's not necessary to do in this PR